homeassistant/components/vacuum/xiaomi_miio.py

```diff
@@ -220,6 +221,9 @@ def device_state_attributes(self):
                     / 3600),
                 ATTR_FILTER_LEFT: int(
                     self.consumable_state.filter_left.total_seconds()
+                    / 3600),
+                ATTR_SENSOR_DIRTY: int(
+                    self.consumable_state.sensor_dirty.total_seconds()
```

Comment:
We should not ever contain a relative time in our attributes. All these total seconds have to go. We should only record timestamps in the state machine.

Reply:
The state of an entity should not change based on just time. It causes a state change every time the entity gets written to the state machine even though nothing has changed.

Reply:
Agreed, I'll look into this next week when I get some time. The attribute should return the timestamp when a change is required, I think.